File: Day-1/solution.js
// var fs = require('fs');

// const data = fs
//   .readFileSync('./input.txt', { encoding: 'utf8' })
//   .split('\r\n')
//   .map(Number);

// // PART ONE
// const part1 = data.reduce((a, c) => {
//   return a + c;
// }, 0);

// console.log('Part 1: ', part1);

// // PART 2
// let frequency = 0;
// let map = new Map();
// let i = 0;

// while (!map.has(frequency)) {
//   map.set(frequency, map.size);
//   frequency += data[i % data.length];
//   i++;
// }

// console.log('Part 2: ', frequency);

const fs = require('fs');
const data = fs
  .readFileSync('./input.txt', 'utf8')
  .split('\r\n')
  .map(Number);

// Part 1
const resultPart1 = data.reduce((a, c) => a + c);
console.log(resultPart1);

// Part 2
const list = new Map();
let freq = 0;
let i = 0;

while (!list.has(freq)) {
  list.set(freq);
  freq += data[i % data.length];
  i++;
}

console.log(freq);

File: Day-1/solution.py
data = [int(x) for x in open('./input.txt').read().splitlines()]
<<<<<<< HEAD

# Part 1
print('Part 1: ', sum(data))

# Part 2
check = set()
current_freq = 0
i = 0

while current_freq not in check:
    check.add(current_freq)
=======
# PART 1
print('Part 1: ', sum(data))

# PART 2
freqs = set()
current_freq = 0
i = 0

while current_freq not in freqs:
    freqs.add(current_freq)
>>>>>>> bd3fceeba71a64ae2980a42185059441cdf311c9
    current_freq += data[i % len(data)]
    i += 1

print('Part 2: ', current_freq)
